src/Algebra/Definitions.agda

```diff
@@ -26,21 +25,122 @@ module Algebra.Definitions
 open import Algebra.Core using (Op₁; Op₂)
 open import Data.Product.Base using (_×_; ∃-syntax)
 open import Data.Sum.Base using (_⊎_)
+open import Function.Base using (flip; id; _∘_; _on_)
+open import Relation.Nullary.Negation.Core using (¬_)
+open import Relation.Unary using (Pred)
+
+------------------------------------------------------------------------
+-- Properties of operations
+
+------------------------------------------------------------------------
+-- A generalisation: because not everything can be defined in terms of
+-- a single relation _≈_, AND because the corresponding definitions in
+-- `Relation.Binary.Core` use *implicit* quantification... so there is
+-- plenty of potential for later/downstream refactoring of this design
+
+module Monotonicity {ℓ₁ ℓ₂} (_≤₁_ : Rel A ℓ₁) (_≤₂_ : Rel A ℓ₂) where
```

Comment:
You've probably realized this, but so it doesn't get forgotten: these definitions don't use `_≈_` but are passed it anyway. We should have files `Algebra.Definitions.Monotonicity` and `Algebra.Definitions.Cancellativity` to avoid this if we keep these

Reply:
Good point, and I had indeed, overlooked this aspect of the design, in favour of trying to work out in eg `Data.Rational.Properties` how to selctively instantiate `Definitions` at various points with relations *other than* equality... on the analogy with `*-Reasoning` modules...